src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8322,4 +8346,217 @@ public void ListPattern_NullTestOnSlice()
             Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[1,2,3]").WithLocation(15, 10)
             );
     }
+
+    [Fact, WorkItem(59465, "https://github.com/dotnet/roslyn/issues/59465")]
+    public void MissingLength()
+    {
+        var source = @"
+_ = new S() is [];
+_ = new S() is [..];
+_ = new S() is [0, .. var x, 1];
+
+struct S
+{
+    public int this[System.Index i] => 0;
+    public int this[System.Range i] => 0;
+}
+";
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics(
+            // (2,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[]").WithArguments("S").WithLocation(2, 16),
+            // (3,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [..];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[..]").WithArguments("S").WithLocation(3, 16),
+            // (4,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [0, .. var x, 1];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[0, .. var x, 1]").WithArguments("S").WithLocation(4, 16)
+            );
+    }
+
+    [Fact, WorkItem(59465, "https://github.com/dotnet/roslyn/issues/59465")]
+    public void MissingLength_WithIntIndexer()
+    {
+        var source = @"
+_ = new S() is [];
+
+struct S
+{
+    public int this[int i] => i;
+}
+";
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics(
+            // (2,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[]").WithArguments("S").WithLocation(2, 16),
+            // (2,16): error CS1503: Argument 1: cannot convert from 'System.Index' to 'int'
```

Comment:
>error CS1503: Argument 1: cannot convert from 'System.Index' to 'int' [](http://example.com/codeflow?start=23&length=69)

This error feels confusing. I have no idea what it is trying to tell me.

Reply:
Agreed the error could be confusing. 
This is the same error you get if you try to index with `System.Index` into this type:
```
_ = new S()[^1]; // error CS1503: Argument 1: cannot convert from 'System.Index' to 'int'

struct S
{
    public int this[int i] => i;
}
```
I'm not planning to address in this PR, feel free to open an issue.